src/solvers/dg/dg_2d.jl

```diff
@@ -588,14 +588,14 @@ function calc_boundary_flux!(cache, t, boundary_conditions,
 
   # Calc boundary fluxes in each direction
   for direction in eachindex(firsts)
-    calc_boundary_flux_by_direction!(surface_flux_values, t, boundary_conditions,
+    calc_boundary_flux_by_direction!(surface_flux_values, t, boundary_condition,
                                      equations, dg, cache,
                                      direction, firsts[direction], lasts[direction])
   end
 end
 
-function calc_boundary_flux!(cache, t, boundary_conditions::NTuple{4,Any},
-                             equations::AbstractEquations{2}, dg::DG) # 4 = 2*ndims
+function calc_boundary_flux!(cache, t, boundary_condition::Union{NamedTuple,Tuple},
```

Comment:
Why the type annotations on `boundary_condition` (out of curiosity)? Are we dispatching on it or is this really just a safeguard against something?

Reply:
We're dispatching here: We have three methods using
```julia
boundary_condition::BoundaryConditionPeriodic # do nothing
boundary_condition # apply a single function to every boundary
boundary_condition::Union{NamedTuple,Tuple} # one function per boundary
```

Reply:
:+1: Thanks for the clarification